packages/react-html/README.md

```diff
@@ -12,112 +12,267 @@ $ yarn add @shopify/react-html
 
-The component will automatically propagate any usage of the `react-helmet` module in your app’s content to manipulate the title or other top level HTML or HEAD attributes.
+This function creates serialize/ deserialize components. These can be used by components that manage a stateful part of the application that must be communicated from the server to the client (for example: async loaded translations, Redux, Apollo, etc).
 
-## Interface
+`createSerializer()` accepts a single string argument for the identifier to use; this will help you find the serialized `script` tag if you need to debug later on. It also accepts a generic type argument for the type of the data that will be serialized/ available after deserialization.
 
-```typescript
-export interface Props {
-  children?: React.ReactNode;
-  styles?: Asset[];
-  scripts?: Asset[];
-  blockingScripts?: Asset[];
-  headData?: {[id: string]: any};
-  data?: {[id: string]: any};
-  hideForInitialLoad?: boolean;
-}
+The function returns a pair of components:
 
-interface Asset {
-  path: string;
-  integrity?: string;
-}
+```tsx
+createSerializer<string>('MyData');
 
-interface Browser {
-  userAgent: string;
-  supported: boolean;
-}
+// Would create the following components:
+
+function Serialize({data}: {data(): string}): null;
+function WithSerialized({
+  children,
+}: {
+  children(data: string | undefined): React.ReactNode;
+}): React.ReactNode;
 ```
 
-### Basic props
+The general pattern for using these components is to render the `WithSerialized` component as the top-most child. Within the render prop, construct whatever stateful store/ manager you need, using the data that was retrieved in cases where the serialization was found (on the browser, or on subsequent server renders). Finally, render the UI that depends on that stateful part, and a `Serialize` component that extracts the part that you you need to communicate between server and client.
 
-Most simple applications will only need these basic properties.
+Here is a complete example, using `@shopify/react-i18n`’s support for async translations as the data that needs to be serialized:
 
-**children**
-The children to be rendered inside the `#app` div.
+```tsx
+import {createSerializer} from '@shopify/react-html';
+import {Provider, Manager} from '@shopify/react-i18n';
 
-**styles**
-Descriptors for any style tags you want to include in the HEAD of the document.
+interface Props {
+  locale: string;
+  children: React.ReactNode;
+}
 
-**scripts**
-Descriptors for any script tags you want to include in your document. All scripts passed to this property will be deferred by appending them to the end of the document. We encourage this as a default, although you may use `blockingScripts` for any scripts that must be included in the HEAD of the document.
+const {Serialize, WithSerialized} = createSerializer<
+  ReturnType<Manager['extract']>
+>('i18n');
+
+export default function I18n({locale, children}: Props) {
```

Comment:
```suggestion
export default function MyComponentThatUsesTranslations({locale, children}: Props) {
```

Might be nice to make it clear that this is a component that uses i18n. On my first pass at reading this I wondered if this name had special meaning based on the imported library and it doesn't

Reply:
It is not a component that uses i18n, it is a component that provides i18n to the rest of the app with the help of the i18n package.